packages/retail-ui-extensions/src/extension-api/NavigationApi/index.ts

```diff
@@ -0,0 +1,18 @@
+interface NavigationApiContent {
+  /** Opens a full screen with the Retail::POS::SmartGrid::Modal extension point */
+  navigateToFullScreenModal(): void;
+
+  /** Dismisses the modal highest on the stack */
+  dismiss(): void;
+}
+
+/**
+ * Access the navigation API to present your app in a full screen modal.
+ */
+export interface NavigationApi {
+  navigation: NavigationApiContent;
+}
+
+export function isNavigationApi(api: any): api is NavigationApi {
```

Comment:
Will not all extensions have access to these APIs? It's hard to tell what the value of these "does this API have this functionality" functions is without knowing the actual API available to POS extensions.